src/components/Profile/components/Identification/components/SocialMediaLinks/components/LinksDialog/index.jsx

```diff
@@ -60,8 +60,8 @@ const LinksDialog = ({ links, createSnackbar, onClose, setLinks }) => {
       console.log('Failed Updates: ', failedUpdates);
       createSnackbar('Failed Updating Link(s)', 'error');
     } else {
-      onClose();
       createSnackbar('Social Media Links Updated', 'success');
+      window.location.reload();
```

Comment:
The issue here is that we want to reload the user's profile now that their links have been updated, right? This is one way to solve that problem that definitely works, but there is another way that might be better.

There is a custom hook, [`useUserActions`](https://github.com/gordon-cs/gordon-360-ui/blob/develop/src/hooks/useUserActions.js), which references the [`UserActionsContext`](https://github.com/gordon-cs/gordon-360-ui/blob/a727d77abcafedcfe2bd109a576a7854388f2fba/src/contexts/UserContext.tsx#L19). The hook returns an object that has an `updateProfile` function which, when called, forces the front-end to refresh the current user's profile information. So, if you call the `useUserActions` hook at the top of this component to get the `updateProfile` function, you can call `updateProfile` on this line instead. That will refresh the user's profile info (including their links), and React will re-render only the components that depend on the user's profile.

This would be a much more performant solution, since we're only reloading the profile and only re-rendering the components that depend on the profile. It would also be easier to understand, so people looking at this code in the future will know what we're trying to achieve.

If you want to look at an example that uses `useUserActions` for a similar purpose, you can look at the [`IDUPloader`](https://github.com/gordon-cs/gordon-360-ui/blob/a727d77abcafedcfe2bd109a576a7854388f2fba/src/views/IDUploader/index.jsx#L54). The `IDUploader` component needs to refresh the user's profile images when they submit an ID image, so it uses the `updateImage` function that is returned from `useUserActions`, but it's the same basic idea as using `updateProfile`.